alarm_trap_sender.cpp

```diff
@@ -50,35 +50,61 @@ AlarmFilter AlarmFilter::_instance;
 
 AlarmTrapSender AlarmTrapSender::_instance;
 
-bool ActiveAlarmList::update(AlarmTableDef& alarm_table_def, const std::string& issuer)
+bool ObservedAlarms::update(AlarmTableDef& alarm_table_def, const std::string& issuer)
 {
-  bool updated = false;
+  bool send_trap = false;
 
   std::map<unsigned int, AlarmListEntry>::iterator it = _index_to_entry.find(alarm_table_def.alarm_index());
-
+  
+  // Enter this if statement if the alarm we are currently raising is being
+  // raised with a non-CLEARED severity. 
   if (alarm_table_def.is_not_clear())
```

Comment:
Agreed. This also means that `update` and `override` are basically the same function, except that the latter doesn't return whether an update happened. If the caller does not care about this, it could just call `update` and ignore the return code, so we can bin `override` entirely.